hydra-node/src/Hydra/Network/Ouroboros.hs

```diff
@@ -192,10 +194,10 @@ withOuroborosNetwork tracer localHost remoteHosts networkCallback between = do
    where
     networkServerTracers =
       NetworkServerTracers
-        { nstMuxTracer = nullTracer
-        , nstHandshakeTracer = contramap TraceHandshake tracer
-        , nstErrorPolicyTracer = contramap TraceErrorPolicy tracer
-        , nstAcceptPolicyTracer = contramap TraceAcceptPolicy tracer
+        { nstMuxTracer = contramap (TraceMux localHost) tracer
+        , nstHandshakeTracer = contramap (TraceHandshake localHost) tracer
+        , nstErrorPolicyTracer = contramap (TraceErrorPolicy localHost) tracer
+        , nstAcceptPolicyTracer = contramap (TraceAcceptPolicy localHost) tracer
```

Comment:
In the ouroboros-network codebase, I've often seen the `WithXXX` pattern for encapsulating traces in a certain context, while avoid to repeat the context on each branch of the trace, like:

```hs
data WithHost trace = WithHost Host trace
```

It keeps the actual trace (here `TraceOuroborosNetwork`) clean, while still providing a nice context to the tracer.